### PR TITLE
Keep PL Logger config in `KrakenTrainer`

### LINKED ISSUE #477: [request] TensorBoard logger should not be (a hyperparameter) in LightningModule
While writing, I may have thought of a workaround. Still, I think this request could be considered and if it is, I can probably create a PR for it. I think the core of my issue is that the TensorBoard logger (info) is given to both the `RecognitionModel` (and `SegmentationModel`) as well as the `KrakenTrainer`, whereas IMHO it should only be a concern of the `KrakenTrainer`.

I would like to train a recognition model with monitoring using TensorBoard and specify extra callbacks, like a [`DeviceStatsMonitor`][dsm]. I also want to specify the name of the model as the `name` argument of the [`TensorBoardLogger`][tbl].
To prevent modifying Kraken's code I copied and modified the `ketos` CLI, which creates the `RecognitionModel` and `KrakenTrainer` with a few parameters from [`pl.Trainer`][plt].

[dsm]: https://lightning.ai/docs/pytorch/stable/api/pytorch_lightning.callbacks.DeviceStatsMonitor.html#pytorch_lightning.callbacks.DeviceStatsMonitor
[tbl]: https://lightning.ai/docs/pytorch/stable/extensions/generated/lightning.pytorch.loggers.TensorBoardLogger.html#lightning.pytorch.loggers.TensorBoardLogger
[plt]: https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#trainer-class-api

Based on https://github.com/mittagessen/kraken/blob/44a9933266a345f8ba1013f4ca5de3ce14f20808/kraken/lib/train.py#L61-L76 I figured I could overwrite the `'tensorboard'` value of `pl_logger` with my own `TensorBoardLogger` with the right parameters and provide it to the `KrakenTrainer` as the `pl_logger` and the `logger` *kwarg*.

However, `pl_logger` in the `ketos train` command is also saved as a hyperparameter and since a `TensorBoardLogger` is not JSON-encodable, the training crashed when the model was to be saved.

A simple workaround appeared to be to remove `pl_logger` from the hyperparameters provided to the `RecognitionModel`. But then I could not monitor the learning rate, as the `LearningRateMonitor` is only added if `pl_logger` is a hyperparameter:
https://github.com/mittagessen/kraken/blob/44a9933266a345f8ba1013f4ca5de3ce14f20808/kraken/lib/train.py#L575-L584

I would like to propose:

- remove `pl_logger` from the training hyperparameters
- add the `LearningRateMonitor` in `KrakenTrainer.__init__`
- optionally add CLI options for the experiment name, logging the graph and creating a `DeviceStatsMonitor` callback

How do others feel about this?

### PR DESCRIPTION
**Pull request by bencomp**
## What this does

This closes #477 by keeping all data concerning PyTorch Lightning's `TensorBoardLogger` in the CLI commands and `KrakenTrainer`, not the Lightning Modules' hyperparameters.
As discussed in #477 this is useful when you create a `KrakenTrainer` yourself and want to give more options to the TensorBoardLogger. If you use the `ketos` command with `--logger tensorboard`, nothing should change.

## Detailed changes

- Remove `pl_logger` from (default) hyperparameters for `RecognitionModel` and `SegmentationModel`;
- Check for existing PL Logger before creating a default `TensorBoardLogger` during initialisation of `KrakenTrainer`;
- Add `LearningRateMonitor` callback during Trainer init instead of Module setup.

## To discuss

A `ValueError` is raised if the value for `pl_logger` is not a Logger or the `tensorboard` string and the `logger` keyword argument is not a Logger. I think an error would be raised in this case later on anyway, when the learning-rate monitor is added, but it felt appropriate to raise here, where the problem is.

## Additional notes

Note that an error will be raised if the TensorBoardLogger is initialised with `log_graph=True`. This happens because the `forward` method in the `RecognitionModel` returns or may return a tuple with `None` as the second element, which interferes with logging the graph of the neural network.

### CODE REVIEW

**Comment by mittagessen** [comment]
Sorry, I renamed the master branch to main yesterday which auto-closed your pull request. I've merged it manually now.

**Comment by bencomp** [comment]
Thanks! I was indeed a bit surprised when it was closed without explanation, and after seeing the reason planned to recreate the PR. So thanks for merging!
